lib/components/tag-chip/index.tsx

```diff
@@ -1,24 +1,20 @@
-import React from 'react';
+import React, { FunctionComponent } from 'react';
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
-import { noop } from 'lodash';
 
-export const TagChip = ({ onSelect = noop, selected, tag: tagName }) => (
+type OwnProps = {
+  onSelect: () => any;
+  selected: boolean;
+  tag: string;
+};
+
+const TagChip: FunctionComponent<OwnProps> = ({ onSelect, selected, tag }) => (
```

Comment:
out of curiosity, why did you rename the `tagName` here while adding types?

if my memory serves me right we renamed the prop inside the component because it read more clearly as "tag name" than "tag" did, which conflates with one of several types of "tag" (`TagEntity`, `Tag`, `TagName`) but in part due to lack of typing around the app we didn't update all the call-sites

Reply:
I'm all for renaming but we should rename the property given to the TagChip

Reply:
> I'm all for renaming 

I might have miscommunicated. This PR does more than add types - it renames an internal variable. That stood out to me, even if I can agree to it. In this case, it seems like we went from something that in the past we made more specific to decrease confusion to something more generic and I didn't see an explanation of why we changed it.

Reply:
I understood what you are saying, I think the wrong place to change the name is in the component. It should be changed everywhere that the component is called.

I agree that changing the variable name was out of scope for this PR.